src/metabase/api/database.clj

```diff
@@ -445,21 +446,44 @@
         fields (readable-fields-only (autocomplete-fields db-id match-string limit))]
     (autocomplete-results tables fields limit)))
 
+(def ^:private autocomplete-matching-options
+  "Valid options for the autocomplete types. Can match on a substring (\"%input%\"), on a prefix (\"input%\"), or reject
+  autocompletions. Large instances with lots of fields might want to use prefix matching or turn off the feature if it
+  causes too many problems."
+  #{:substring :prefix :off})
+
+(defsetting sql-editor-autocomplete-match-style
```

Comment:
```suggestion
(defsetting native-query-autocomplete-match-style
```
```suggestion
    (str "Matching style for native query's autocomplete. Can be \"substring\", \"prefix\", or \"off\". "
```

It works for mongo too, so I think sql editor might be misleading.

Also "native query" match with our naming for the main entities (question, native query, dashboard, collection), which I think makes it easier for users to know what this is about.

Reply:
good call. We've updated